src/pages/ProductList/index.js

```diff
@@ -1,52 +1,154 @@
-import { useState, useCallback } from "react";
+import { useState, useCallback, useMemo, useEffect } from "react";
 import { Products, Loader, SidebarItem, Pagination } from "../../components";
-import { useProducts } from "../../hooks";
+import { useQuery, useRequests } from "../../hooks";
 import classNames from "classnames";
-import categoriesData from "../../mocks/en-us/product-categories.json";
-import "./ProductList.scss";
+import { Button } from "../../styled";
+import {
+  WrapperProductList,
+  Sidebar,
+  SidebarTitle,
+  ClearFilters,
+  Content,
+} from "./styled";
+
+let PageSize = 12;
 
 function ProductList() {
-  const [filters, setFilters] = useState([]);
-  const [products, isLoading] = useProducts(filters);
+  let query = useQuery();
+  let categorySlug = query.get("category");
+
+  const requests = useMemo(
+    () => [
+      {
+        name: "productCategories",
+        params: {
+          q: '[[at(document.type, "category")]]',
+          pageSize: 30,
+        },
+      },
+      {
+        name: "productsList",
+        params: {
+          q: '[[at(document.type, "product")]]',
+          pageSize: 100,
+        },
+      },
+    ],
+    []
+  );
+
+  const {
+    data: { productCategories, productsList },
+    isLoading,
+  } = useRequests(requests);
+
+  const [products, setProducts] = useState([]);
+  const [activeFilters, setActiveFilters] = useState([]);
+
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const currentProducts = useMemo(() => {
+    const firstPageIndex = (currentPage - 1) * PageSize;
+    const lastPageIndex = firstPageIndex + PageSize;
+    return products.slice(firstPageIndex, lastPageIndex);
+  }, [currentPage, products]);
+
+  useEffect(() => {
+    if (isLoading || !categorySlug) {
```

Comment:
If you want to do something when this is false, maybe change the logic instead like:

```javascript
if (!isLoading && categorySlug) {
   ...your logic here
}
```

This way you avoid those empty returns.